collector/thermalzone.go

```diff
@@ -0,0 +1,105 @@
+package collector
+
+import (
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	Factories["thermalzone"] = NewThermalZoneCollector
+}
+
+// A ThermalZoneCollector is a Prometheus collector for WMI Win32_PerfRawData_Counters_ThermalZoneInformation metrics
+type ThermalZoneCollector struct {
+	HighPrecisionTemperature *prometheus.Desc
+	PercentPassiveLimit      *prometheus.Desc
+	Temperature              *prometheus.Desc
+	ThrottleReasons          *prometheus.Desc
+}
+
+// NewThermalZoneCollector ...
+func NewThermalZoneCollector() (Collector, error) {
+	const subsystem = "thermalzone"
+	return &ThermalZoneCollector{
+		HighPrecisionTemperature: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "high_precision_temperature"),
+			"(HighPrecisionTemperature)",
+			nil,
+			nil,
+		),
+		PercentPassiveLimit: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "percent_passive_limit"),
+			"(PercentPassiveLimit)",
+			nil,
+			nil,
+		),
+		Temperature: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "temperature"),
+			"(Temperature)",
+			nil,
+			nil,
+		),
+		ThrottleReasons: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "throttle_reasons"),
```

Comment:
Have you found any documentation on what values this can have?

Reply:
per [this documentation](https://wutils.com/wmi/root/cimv2/win32_perfrawdata_counters_thermalzoneinformation/#temperature_properties), it holds a uint32. [This page](https://docs.microsoft.com/en-us/windows-hardware/design/device-experiences/examples--requirements-and-diagnostics) shows the codes to their corresponding meaning. I'll add some documentation for that.